aether-ui/aether/ui/api/views.py

```diff
@@ -70,6 +71,33 @@ def delete_artefacts(self, request, pk=None, *args, **kwargs):
 
         return Response(artefacts_result)
 
+    @action(detail=True, methods=['put'], url_path='rename')
+    def rename(self, request, pk=None, *args, **kwargs):
+        pipeline = self.get_object_or_404(pk=pk)
+        new_name = request.data.get('name', pipeline.name)
+        pipeline.name = new_name
+        pipeline.save()
+        if pipeline.mappingset:
```

Comment:
Do we need to update the kernel "mappingset" in this method? :thinking:

Reply:
Yes, I agreed with @adamvert to propagate to kernel immediately if the pipeline is already published.